crates/matrix-sdk/src/test_utils/mocks.rs

```diff
@@ -316,6 +321,166 @@ impl MatrixMockServer {
         MockEndpoint { mock, server: &self.server, endpoint: RoomSendEndpoint }
     }
 
+    /// Creates a prebuilt mock for sending an event with a specific type in a
+    /// room.
+    ///
+    /// Note: works with *any* room.
+    ///
+    /// Similar to: [`MatrixMockServer::mock_room_send`]
+    ///
+    /// # Examples
+    ///
+    /// see also [`MatrixMockServer::mock_room_send`] for more context.
+    ///
+    /// ```
+    /// # tokio_test::block_on(async {
+    /// use matrix_sdk::{ruma::{room_id, event_id}, test_utils::mocks::MatrixMockServer};
+    /// use serde_json::json;
+    ///
+    /// let mock_server = MatrixMockServer::new().await;
+    /// let client = mock_server.client_builder().build().await;
+    ///
+    /// mock_server.mock_room_state_encryption().plain().mount().await;
+    ///
+    /// let room = mock_server
+    ///     .sync_joined_room(&client, room_id!("!room_id:localhost"))
+    ///     .await;
+    ///
+    /// let event_id = event_id!("$some_id");
+    /// mock_server
+    ///     .mock_room_send_for_type("m.room.message".into())
+    ///     .ok(event_id)
+    ///     .expect(1)
+    ///     .mount()
+    ///     .await;
+    ///
+    /// let responseNotMocked = room.send_raw("m.room.reaction", json!({ "body": "Hello world" })).await;
+    /// // The `m.room.reaction` event type should not be mocked by the server.
+    /// assert!(responseNotMocked.is_err());
+    ///
+    ///
+    /// let response = room.send_raw("m.room.message", json!({ "body": "Hello world" })).await?;
+    /// // The `m.room.message` event type should be mocked by the server.
+    /// assert_eq!(
+    ///     event_id,
+    ///     response.event_id,
+    ///     "The event ID we mocked should match the one we received when we sent the event"
+    /// );
+    /// # anyhow::Ok(()) });
+    /// ```
+    pub fn mock_room_send_for_type(
+        &self,
+        event_type: MessageLikeEventType,
+    ) -> MockEndpoint<'_, RoomSendEndpoint> {
+        let mock = Mock::given(method("PUT"))
+            .and(path_regex(format!(r"^/_matrix/client/v3/rooms/.*/send/{}", event_type)))
+            .and(header("authorization", "Bearer 1234"));
+        MockEndpoint { mock, server: &self.server, endpoint: RoomSendEndpoint }
+    }
```

Comment:
This should not be implemented this way: instead there should be a single `mock_room_send`, and the `MockEndpoint<RoomSendEndpoint>` impl block should have a method to limit `for_type(self, event_type: MessageLikeEventType) -> Self`. Otherwise we're losing compsability at the endpoint level, and risk duplicating endpoints lots of time, which we'd like to avoid.